### PR TITLE
[Merged by Bors] - chore: address porting note about type-ascribing `Quotient.mk`

### PR DESCRIPTION
**Pull request by Vierkantor**
It no longer appears to be the case that we need a type ascription on `Quotient.mk` for performance reasons: in fact according to `count_heartbeats` the number of heartbeats goes down very slightly on basically every declaration if we remove them. So remove the porting note and the type ascriptions.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/Quotient.lean
Comment (grunweg) [inline Mathlib/LinearAlgebra/Quotient.lean:67]: Do you still need the type ascription for `mk x`? Because the PR descriptions sounds like you could remove those as well...

File: Mathlib/LinearAlgebra/Quotient.lean
Comment (grunweg) [inline Mathlib/LinearAlgebra/Quotient.lean:70]: Same here.

File: Mathlib/LinearAlgebra/Quotient.lean
Comment (grunweg) [inline Mathlib/LinearAlgebra/Quotient.lean:93]: And here, and in the two declarations below.

**Review by grunweg — APPROVED** [review]
Trusting that you tested this, this change is a clear improvement in readability. Thanks!
I have one small question, otherwise LGTM.

File: Mathlib/LinearAlgebra/Quotient.lean
Comment (Vierkantor) [reply Mathlib/LinearAlgebra/Quotient.lean:67]: In these cases yes, because otherwise it's not clear which quotient is being taken.

**Comment by grunweg** [comment]
In this case: 
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**
